### PR TITLE
RaR: Arella Salvatore

### PR DESCRIPTION
**Pull request by jwarwick**
Needs a test.

### CODE REVIEW

**Comment by NoahTheDuke** [comment]
That looks good. Mind renaming `params` to `args`?

File: src/clj/game/cards/upgrades.clj [hunk 1]
```diff
@@ -20,6 +20,25 @@
                                  :prompt "Rez ICE with rez cost lowered by 3?" :priority 2
                                  :yes-ability {:effect (effect (rez-cost-bonus -3) (rez target))}}}}}
 
+   "Arella Salvatore"
+   {:events
+    {:agenda-scored
+     {:req (req (= (:previous-zone target) (:zone card)))
+      :interactive (req true)
+      :optional {:prompt "Install a card from HQ?"
```
Comment (nealterrell) [inline src/clj/game/cards/upgrades.clj]: Similar to #3723, I prefer optional-then-card-selection prompts to be a single selection prompt, with the `:cancel-effect` being used as the `:no-ability` equivalent when needed. The prompt can then read "Install a card from HQ with Arella Salvatore?".

File: src/clj/game/cards/upgrades.clj [hunk 2]
```diff
@@ -20,6 +20,24 @@
...
                                  :yes-ability {:effect (effect (rez-cost-bonus -3) (rez target))}}}}}
 
+   "Arella Salvatore"
+   {:events
+    {:agenda-scored
+     {:req (req (and (= (:previous-zone target) (:zone card))
+                     (some #(corp-installable-type? %) (:hand corp))))
+      :interactive (req true)
+      :prompt "Select a card to install with Arella Salvatore"
+      :choices (req (cancellable (filter #(corp-installable-type? %) (:hand corp))))
```
[the remark this answers is not in the document]
Comment (jwarwick) [reply src/clj/game/cards/upgrades.clj]: I went back and forth between the two. I thought `Done` was less clear than `Cancel`. But I don't have a strong preference.

[the remark this answers is not in the document]
Comment (Saintis) [reply src/clj/game/cards/upgrades.clj]: I think target prompt makes more sense as well.

**Pull request merged**